tests/ssl-opt.sh

```diff
@@ -1289,6 +1289,26 @@ run_tests_memory_after_hanshake()
     run_test_memory_after_hanshake_with_mfl 512 "$MEMORY_USAGE_MFL_16K"
 }
 
+run_test_tls13_compat()
```

Comment:
We have some automation that collects the available test cases. See `tests/scripts/check_test_cases.py`. This automation parses `ssl-opt.sh` to look for `run_test` calls. It doesn't support `compat.sh` yet.

We currently only use this automation to generate an informative report about tests cases that are never executed (done by `tests/scripts/analyze_outcomes.py`).

It's currently not critical if this automation misses some tests, but it would be better not to add more problematic cases, like here where the automation doesn't know about the new tests.

Reply:
How about add `source tests/ssl-opt-test-cases/*.sh` ? Is it more friendly for `tests/scripts/check_test_cases.py` ? @gilles-peskine-arm

Reply:
I'm not sure what you mean. Do you mean generating a `.sh` file and having `check_test_cases.py` parse it? Yes, that would work.

Yes, there are some test case generators already, which `check_test_cases.py` doesn't support. But it would be nice not to add more.

Reply:
>I'm not sure what you mean. Do you mean generating a .sh file and having check_test_cases.py parse it? Yes, that would work.

Yes. That's my meaning. I will upload  output of `generate_tls13_compat_tests.py` .

Reply:
Done. Please review

Reply:
The shell version looks different. I always fail at `all_fbsd_test_cmake_out_of_box_build` . `ssl_opt.sh` line 36 is to fix that. 

To change shebang line only can not pass the test, it reports `Operation not permission` . And `/bin/sh` does not support `source` command.  So I revert shebang line and add some code to run scripts with `bash`. 

It looks too complex, but it works now.